kmeans_functions.py

```diff
@@ -1,31 +1,32 @@
 import numpy as np
-import scipy.spatial.distance as ssdist
 import initialization as init
+import scipy.spatial.distance as ssdist
+import numbers
 
 
-def initialization(n_clusters, initialize, x):
+def initialization(n_clusters, initialize, x, random_state, metric):
     # this method initializes the first set of labels
     if initialize == 'random':
-        centroids = init.random_init(x, n_clusters)
+        centroids = init.random_init(x, n_clusters, random_state)
     elif initialize == 'k-means++':
-        centroids = init.kpp_init(x, n_clusters)
+        centroids = init.kpp_init(x, n_clusters, metric)
     return centroids
 
 
-def labeling(centroids, x):
+def labeling(centroids, x, metric):
     # this method creates labels for every position in training data based on current centroids
-    distance = fix_the_distance(x, centroids, 'correlation')
+    distance = compute_distance(x, centroids, metric)
     labels = np.argmin(distance, axis=1)
     return labels
 
 
-def compute_centroids(n_clusters, centroids, labels, n_iter, max_iter, x):
+def compute_centroids(n_clusters, centroids, labels, n_iter, max_iter, x, metric):
     while n_iter < max_iter:
         if n_iter > 0:
             prev_centroids = centroids
         centroids = _centroid_by_means(n_clusters, centroids, labels, x)
-        labels = labeling(centroids, x)
-        if n_iter > 5 and np.all(centroids == prev_centroids):
+        labels = labeling(centroids, x, metric)
+        if n_iter > 0 and np.all(centroids == prev_centroids):

```

Comment:
It's better to compare previous labels - less numbers to compare. Additionally comparison would be done on `int` instead of `float`.